src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala

```diff
@@ -24,163 +29,187 @@ case class CromwellRuntimeAttributes(docker: Option[String],
-    val Preemptible = 0
-    val Zones = Vector("us-central1-a")
-  }
-
-  private def validateRuntimeAttributes(attributeMap: AttributeMapTrait): ValidationNel[String, CromwellRuntimeAttributes] = {
-    val docker = attributeMap.get(DOCKER)
-    val zones = attributeMap.get(ZONES) map { _.split("\\s+").toVector } getOrElse Defaults.Zones
+  val LocalizationDisk = LocalDisk(LocalDiskName, DiskType.SSD, sizeGb=10)
+
+  private val keys = Set("cpu", "disks", "docker", "zones", "continueOnReturnCode", "failOnStderr", "preemptible", "memory")
+
+  private val defaultValues = Map(
+    "cpu" -> WdlInteger(1),
+    "disks" -> WdlString(LocalizationDisk.toString),
+    "zones" -> WdlString("us-central1-a"),
+    "continueOnReturnCode" -> WdlInteger(0),
+    "failOnStderr" -> WdlBoolean.False,
+    "preemptible" -> WdlInteger(0),
+    "memory" -> WdlString("2GB")
+  )
+
+  private def defaultValues(workflowOptions: WorkflowOptions): Map[String, WdlValue] = {
+    keys.collect({
+      case k if workflowOptions.getDefaultRuntimeOption(k).isSuccess =>
+        k -> WdlString(workflowOptions.getDefaultRuntimeOption(k).get)
+      case k if defaultValues.contains(k) =>
+        k -> defaultValues.get(k).get
+    }).toMap
+  }
+
+  val defaults: CromwellRuntimeAttributes = {
+    // .get below because we assume that the default values are valid
+    // There's a unit test in CromwellRuntimeAttributeSpec to test these defaults
+    validateRuntimeAttributes(defaultValues).get
+  }
+
+  private def validateRuntimeAttributes(attributes: Map[String, WdlValue]): Try[CromwellRuntimeAttributes] = {
+    val attributeMap = attributes.collect({ case (k, v) if Try(RuntimeKey.from(k)).isSuccess => RuntimeKey.from(k) -> v })
+    val docker = validateDocker(attributeMap.get(DOCKER))
+    val zones = validateZone(attributeMap.get(ZONES))
     val failOnStderr = validateFailOnStderr(attributeMap.get(FAIL_ON_STDERR))
-    val continueOnReturnCode = validateContinueOnReturnCode(attributeMap.getSeq(CONTINUE_ON_RETURN_CODE))
+    val continueOnReturnCode = validateContinueOnReturnCode(attributeMap.get(CONTINUE_ON_RETURN_CODE))
     val cpu = validateCpu(attributeMap.get(CPU))
     val disks = validateLocalDisks(attributeMap.get(DISKS))
     val preemptible = validatePreemptible(attributeMap.get(PREEMPTIBLE))
-    val memory = validateMemory(attributeMap.get(MEMORY))
+    val memory = RuntimeAttributes.validateMemoryValue(attributeMap.getOrElse(MEMORY, WdlString(s"${defaults.memoryGB} GB"))) match {
+      case scala.util.Success(x) => x.to(MemoryUnit.GB).amount.successNel
+      case scala.util.Failure(x) => x.getMessage.failureNel
+    }
 
-    (failOnStderr |@| continueOnReturnCode |@| cpu |@| preemptible |@| disks |@| memory) {
-      CromwellRuntimeAttributes(docker, zones, _, _, _, _, _, _)
+    (docker |@| zones |@| failOnStderr |@| continueOnReturnCode |@| cpu |@| preemptible |@| disks |@| memory) {
+      new CromwellRuntimeAttributes(_, _, _, _, _, _, _, _)
```

Comment:
Shouldn't it be a Map? It contains 8 attributes and I think with current implementation it will tend to grow...
It would be easier if runtime attributes is a map and then keys from that map are verified and validated by each backend. Each backend also can contains or/and set default values.
The other thing you gain with this is that the Cromwell-engine is not modified so often.